src/componets/ChooseProductForm/ChooseProductForm.jsx

```diff
@@ -2,26 +2,19 @@ import React from 'react'
 
 const ChooseProductForm = props => {
   return (
-    <>
-      <form onSubmit={props.handleProductEnterClick}>
-        <div className="dialog-board">
-          <span>
-            {props.errorMessageIncorrect
-              ? props.errorMessageIncorrect
-              : props.balance === 0
-              ? '>'
-              : props.selectedProduct
-              ? 'Success'
-              : 'Choose product'}
-          </span>
-        </div>
-        <input
-          onInput={props.productInputValue}
-          ref={props.productInputValueRef}
-          disabled={props.selectedProduct || props.balance === 0}
-        />
-      </form>
-    </>
+    <form onSubmit={props.handleProductEnterClick}>
+      <div className="dialog-board">
+        <span>
+          {props.errorMessageIncorrect ||
+            (props.balance === 0 ? '>' : props.selectedProduct ? 'Success' : 'Choose product')}
```

Comment:
There is still a nested ternary. 😕

Reply:
rewrite on ifs

https://github.com/AlexandrVinogradov/vending-machine-react/blob/b04fd95033283e4691ca7dce340f15cdd90a716d/src/componets/InsertBanknoteForm/InsertBanknoteForm.jsx#L44